mpt-witness-generator/witness/leaf.go

```diff
@@ -418,42 +418,46 @@ func prepareStorageLeafInfo(row []byte, valueIsZero, isPlaceholder bool) ([]byte
 	key := make([]byte, valueLen)
 	value := make([]byte, valueLen)
 
-	var setValue = func(keyLen, offset byte) {
+	var setKeyValue = func(keyLen, offset byte) {
 		if !isPlaceholder {
 			valueRlp = row[keyLen+offset : keyLen+offset+valueRlpLen]
 			if !valueIsZero {
 				copy(value, row[keyLen+offset+valueRlpLen:])
 			}
 		} else {
+			// If placeholder, we leave the value to be 0.
 			valueRlp = []byte{0}
+			// keyRlp holds the RLP of the whole leaf:
+			// - always 192 because when there is a placeholder, the value occupies only 1 byte (can't be 248)
+			// - (keyLen + 1) is the number of key bytes 
+			// - 1 is the number of value bytes
+			keyRlp[0] = 192 + keyLen + 2
 		}
 	}
 
 	if len(row) < 32 { // the node doesn't get hashed in this case
+		keyRlpLen = 1
+		keyRlp = make([]uint8, keyRlpLen)
+		copy(keyRlp, row[:keyRlpLen])
+
```

Comment:
you can add `valueRlpLen = 1` here and remove it from the next lines

Reply:
The line `setKeyValue(keyLen, offset)` could also be moved at the very end of the function (and removed from all if/else conditions).